integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingSenderTestCase.java

```diff
@@ -0,0 +1,246 @@
+        URL endpoint = new URL(getProxyServiceURLHttp("NotFoundEndpointProxy"));
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("404"), "Expected 404, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 404 Not Found"),
+                "Required payload not found");
+    }
+
+    @Test(groups = {"wso2.esb"},
+            description = "Test invoking conflict http status endpoint with blocking call", enabled = true)
+    public void testConflictHttpStatusEndpoint() throws Exception {
+        URL endpoint = new URL(getProxyServiceURLHttp("ConflictEndpointProxy"));
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("409"), "Expected 409, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 409 Conflict"),
+                "Required payload not found");
+    }
+
+
+    @Test(groups = {"wso2.esb"},
+            description = "Test invoking server error http status endpoint with blocking call", enabled = true)
+    public void testServerErrorHttpStatusEndpoint() throws Exception {
+        URL endpoint = new URL(getProxyServiceURLHttp("ServerErrorEndpointProxy"));
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("500"), "Expected 500, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 500 Internal Server Error"),
+                "Required payload not found");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        super.cleanup();
+    }
+
+    public static HttpResponse sendRequest(URL endpoint, Map<String, String> headers)
+            throws AutomationFrameworkException, IOException {
+
+        HttpURLConnection urlConnection = null;
+
+        try {
+            urlConnection = (HttpURLConnection) endpoint.openConnection();
+            try {
+                urlConnection.setRequestMethod("GET");
+            } catch (ProtocolException e) {
+                throw new AutomationFrameworkException("Shouldn't happen: HttpURLConnection doesn't support GET ?? " +
+                        e.getMessage(), e);
```

Comment:
Do we need `e.getMessage()` since we are adding `e` as the cased by exception?

Reply:
Removing `e.getMessage()` in commit [f23167f](https://github.com/wso2/product-ei/pull/3468/commits/f23167fede3bebc4cb3dba75083e82c95eab860c) as `e` prints the whole stack trace.